src/libs/types-rs/src/k8s.rs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2019-2021 Alibaba Cloud
+// Copyright (c) 2019-2021 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::path::Path;
+
+// K8S_EMPTY_DIR is the k8s specific path for `empty-dir` volumes
+const K8S_EMPTY_DIR: &str = "kubernetes.io~empty-dir";
+
+/// Check whether the path is a K8S empty directory.
+///
+/// For a K8S EmptyDir, Kubernetes mounts
+/// "/var/lib/kubelet/pods/<id>/volumes/kubernetes.io~empty-dir/<volumeMount name>"
+/// to "/<mount-point>".
```

Comment:
It's risky to depend on kubelet internal implementation details. But we haven't found a clean way to figure out the EmptyDir medium type from a oci::spec::Mount object.

Reply:
> It's risky to depend on kubelet internal implementation details. But we haven't found a clean way to figure out the EmptyDir medium type from a oci::spec::Mount object.

No doubt - this is yet another example of OCI not giving us information that Kata needs but runc does not.  Is there an equivalent problem in the existing Go code?  How does it handle it?  Do we require this information to work, or only as an optimization?

Reply:
It's an optimization. By default, all volumes will be setup on host and passed to guest by virtio-9p/fs. But that's not performance friend, so we have some done some optimization to move the empty-dir from host into guest.

Reply:
Ok.  If it's just an optimization, and requires this extremely unsafe heuristic, it should be simply left out of the first cut.  It can be added back later, with specific rationales and debates about the performance / risk tradeoff

Reply:
It's a critical optimization, 20-50% performance difference for some workloads. And it may be needed by confidential containers too. But it's ok to hold the k8s related code.

Reply:
Agreed -- there would be functional gaps between the Go and Rust shim implementations without this. We also use volume type to understand if inotify support should be 'emulated'. See https://github.com/kata-containers/kata-containers/blob/main/docs/design/inotify.md

Reply:
Huh.  I hadn't realized the Go runtime was already doing this.  I guess it makes sense to keep it then, horrible as it is.